### ISSUE TITLE
Plans for OA 3.1

### ISSUE THREAD

**Issue by CR4567** [rswag/rswag#464]
I know that this repo is not fully managed anymore. But does anyone has plans to work on support for OA 3.1?
If there will be others, I would also try to do as much as possible.

**Comment by oblakeerickson** [comment]
Yes. I'm on board for getting rswag to support OA 3.1. Any help would be greatly appreciated :)

**Comment by BookOfGreg** [comment]
:+1: Anyone looking to help will also get my attention.

**Comment by joelbiffin** [comment]
I'd be happy to contribute to this

**Comment by tom-quiltt** [comment]
I will be happy to contribute to this. Especially looking to support `New top-level element for describing Webhooks that are registered and managed out of band`.

@BookOfGreg or anyone, is there a list of features that potential contributors can look into?

**Comment by jtannas** [comment]
Heyo - It's mostly @BookOfGreg, @sergioisidoro, and myself trying to revive this repo.
Right now my focus is on getting through the PR backlog before starting on anything major so that we're not losing existing contributions to merge conflicts. I'm moving Monday though so probably not much is going to happen for a week. Help with PR reviews would be really appreciated.

---

My rough plan is:
1. get through the existing PR backlog so that we're not losing existing contributions to merge conflicts
2. Make the repo easier to maintain
    - rubocop autocorrect & enforce via CI to cut down on style PRs & PR comments
    - put in a Cypress E2E test suite for more confidence when making changes
    - drop support for Ruby 2.6 (EOL) and RSpec 2 (v3 was released in 2014)
    - enforce code & branch test coverage in CI? :thinking: 
3. Bugfixes & Features based on community feedback
    - Full OAS 3.0 / 3.1 support seem high on that list
    - Response examples are something that show up frequently in issues
    - ???

**Comment by romanblanco** [comment]
Will leave this here: https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0

**Comment by pboling** [comment]
> drop support for Ruby 2.6 (EOL) and RSpec 2 (v3 was released in 2014)

Just wanted to clarify - the internal specs of the project are already expecting RSpec 3.  It is support for end-users with RSpec 2 suites that is to be dropped, correct?

**Comment by pboling** [comment]
> rubocop autocorrect & enforce via CI to cut down on style PRs & PR comments

I have a gem for that, which is easy to integrate - https://rubocop-lts.gitlab.io/

> enforce code & branch test coverage in CI? 🤔

I have a gem for that which is easy to integrate - https://gitlab.com/kettle-rb/kettle-soup-cover

I'll work on submitting PRs to show what integration with each would look like.

**Comment by romanblanco** [comment]
@pboling Right, FYI, the changes dropping RSpec v2 are already commited in the [`v3`](https://github.com/rswag/rswag/compare/master...v3) branch: https://github.com/rswag/rswag/commit/31861187a47e31e653b012b09c905b1bcffb8df4

**Comment by pboling** [comment]
I think this issue still has value, at least until a milestone is created for the release goals listed above.

**Comment by RobinShift4** [comment]
Is there any way of working around this?  What would you need to see changed in code to support OA 3.1?

**Comment by pboling** [comment]
Why is stale bot considering issues that are part of a milestone? Seems incongruous.
